### PR TITLE
Implement basic income controls

### LINKED ISSUE #521: Implement basic income controls
Should allow policies specifying:
* Basic income amounts per adult, senior (SP age) and child
* Child age threshold, plus whether children can receive Child Benefit and basic income
* Phase-out start threshold and rate
* Inclusion of basic income in benefit means tests

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Fixes #521

@MaxGhenis WDYT? I can add tests too

### CODE REVIEW

File: openfisca_uk/parameters/misc/basic_income/child_age.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+description: Age at which a person stops receiving the child basic income and begins receiving the adult basic income.
+values:
+  2021-01-01: 18
+metadata:
+  unit: year
+  label: Adult basic income theshold
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/misc/basic_income/child_age.yaml]:
```suggestion
  label: Adult basic income threshold
```

File: openfisca_uk/parameters/misc/basic_income/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+child:
+  description: Basic income payment to children.
+  values:
+    2021-01-01: 0
+  metadata:
+    unit: currency-GBP
+    period: week
+    label: Child basic income
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/misc/basic_income/amount.yaml]: don't these need a `name` field?

**Review by MaxGhenis — COMMENTED** [review]
Given the duplication with the UBIs, what about replacing/aliasing the UBI Center `UBI`s with these `basic_income`s, and allow the means-testing on it?

If so, how about locating them in e.g. `parameters/contrib/ubi_center/basic_income`?

File: openfisca_uk/parameters/misc/basic_income/amount.yaml (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_uk/parameters/misc/basic_income/amount.yaml]: Knew I'd forgotten something, thanks

File: openfisca_uk/parameters/contrib/ubi_center/basic_income/child_age.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+description: Age at which a person stops receiving the child basic income and begins receiving the adult basic income.
+values:
+  2021-01-01: 18
+metadata:
+  unit: year
+  label: Adult basic income threshold
+  name: bi_child_age
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/contrib/ubi_center/basic_income/child_age.yaml]: Shouldn't this be `adult_age`?

File: openfisca_uk/variables/finance/benefit/family/JSA_income.py [hunk 4]
```diff
@@ -115,6 +115,10 @@ def formula(benunit, period, parameters):
             "property_income",
             "pension_income",
         ]
+        if parameters(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/finance/benefit/family/JSA_income.py]: Aesthetic - the indentation mix looks awkward to me (below too please)
```suggestion
        bi = parameters(period).contrib.ubi_center.basic_income
        if bi.include_in_means_tests:
```

File: openfisca_uk/variables/hmrc/child_benefit.py [hunk 5]
```diff
@@ -51,6 +51,8 @@ class child_benefit_respective_amount(Variable):
 
     def formula(person, period, parameters):
         eligible = person("is_child_or_QYP", period)
+        if parameters(period).contrib.ubi_center.basic_income.withdraw_cb:
+            eligible *= person("basic_income", period) == 0
```
Comment (MaxGhenis) [inline openfisca_uk/variables/hmrc/child_benefit.py]:
```suggestion
            eligible &= (person("basic_income", period) == 0)
```

**Review by MaxGhenis — COMMENTED** [review]
Yeah I do think a test or two would be helpful

**Comment by nikhilwoodruff** [comment]
@MaxGhenis adapted the CLI tools from openfisca-us so we should now be able to add reform tests in the same way. I've added tests to cover the basic income parameters to this PR.

**Pull request merged**